benches/trap.rs

```diff
@@ -0,0 +1,146 @@
+    for num_threads in vec![1, 2, 4, 8, 16, 32] {
+        group.throughput(Throughput::Elements(num_threads));
+        group.bench_with_input(
+            BenchmarkId::from_parameter(num_threads),
+            &num_threads,
+            |b, &num_threads| {
+                let engine = Engine::default();
+                let module = module(&engine, 10).unwrap();
+
+                b.iter(|| {
+                    let threads = (0..num_threads)
+                        .map(|_| {
+                            std::thread::spawn({
+                                let engine = engine.clone();
+                                let module = module.clone();
+                                move || {
+                                    const ITERS: u32 = 1000;
+                                    for _ in 0..ITERS {
+                                        let mut store = Store::new(&engine, ());
+                                        let instance =
+                                            Instance::new(&mut store, &module, &[]).unwrap();
+                                        let f = instance
+                                            .get_typed_func::<(), (), _>(&mut store, "")
+                                            .unwrap();
+                                        assert!(f.call(&mut store, ()).is_err());
+                                    }
+                                }
+                            })
+                        })
+                        .collect::<Vec<_>>();
+                    threads.into_iter().for_each(|t| t.join().unwrap());
+                });
+            },
+        );
+    }
+
+    group.finish();
+}
+
+fn bench_many_modules_registered_traps(c: &mut Criterion) {
+    let mut group = c.benchmark_group("many-modules-registered-traps");
+
+    for num_modules in vec![1, 8, 64, 512, 4096] {
+        group.throughput(Throughput::Elements(num_modules));
+        group.bench_with_input(
+            BenchmarkId::from_parameter(num_modules),
+            &num_modules,
+            |b, &num_modules| {
+                let engine = Engine::default();
+                let modules = (0..num_modules)
+                    .map(|_| module(&engine, 10).unwrap())
+                    .collect::<Vec<_>>();
+
+                b.iter_custom(|iters| {
+                    let mut total = std::time::Duration::default();
+
+                    for _ in 0..iters {
+                        let mut store = Store::new(&engine, ());
+                        let instance =
+                            Instance::new(&mut store, modules.last().unwrap(), &[]).unwrap();
+                        let f = instance
+                            .get_typed_func::<(), (), _>(&mut store, "")
+                            .unwrap();
+
+                        let start = std::time::Instant::now();
+                        assert!(f.call(&mut store, ()).is_err());
+                        total += start.elapsed();
+                    }
```

Comment:
How come this is re-instantiating each time? Could the same instance be used to use `b.iter(|| ...)`?

Reply:
Switched to one instance/store.